addons/mail/models/discuss/discuss_channel.py

```diff
@@ -463,11 +463,15 @@ def get_field_name(field_description):
                 return field_description.field_name
             return field_description
 
+        def get_field_value(channel, field_description):
+            access_with_sudo = isinstance(field_description, Store.Attr) and field_description.sudo
+            return channel.sudo(access_with_sudo)[get_field_name(field_description)]
+
         def get_vals(channel):
             return {
                 subchannel: {
                     get_field_name(field_description): (
-                        channel[get_field_name(field_description)],
+                        get_field_value(channel, field_description),
```

Comment:
This should probably call `_get_value` from Attr directly instead.

This would also fix other potential issues, such as if Attr does more computation than just reading the fields (lambda value, etc).